### PR TITLE
Proxito: allow overlapping public and external version domains

### PR DESCRIPTION
**Pull request by stsewd**
During development we use overlapping domains:

- devthedocs.org
- build.devthedocs.org

Basically now the suspicious hostname checks are done after the public and external version domains have been tried, and before resolving a custom domain.

Closes https://github.com/readthedocs/readthedocs.org/pull/10241

### CODE REVIEW

**Comment by benjaoming** [comment]
@stsewd is this an alternative to https://github.com/readthedocs/readthedocs.org/pull/10241 ?

**Comment by stsewd** [comment]
> @stsewd is this an alternative to #10241 ?

yeah, it is

File: readthedocs/core/unresolver.py [hunk 1]
```diff
@@ -445,45 +445,38 @@ def unresolve_domain(self, domain):
...
+            project_slug = subdomain
+            log.debug("Public domain.", domain=domain)
+            return UnresolvedDomain(
+                source_domain=domain,
+                source=DomainSourceType.public_domain,
+                project=self._resolve_project_slug(project_slug, domain),
+            )
+
+        # Serve from the RTD_EXTERNAL_VERSION_DOMAIN, ensuring it looks like
+        # `project--version.RTD_EXTERNAL_VERSION_DOMAIN`.
```
Comment (humitos) [inline readthedocs/core/unresolver.py:459]: Good comment! 💯

**Review by humitos — APPROVED** [review]
Thanks for doing this work 🙏🏼

**Pull request merged**